### ISSUE TITLE
Convert SemVer, MediaType and DateTime into Properties

### ISSUE THREAD

**Issue by armintaenzertng** [spdx/spdx-3-model#368]
Currently, [SemVer](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/SemVer.md), [MediaType](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/MediaType.md) and [DateTime](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/DateTime.md) are treated as classes that are subclasses of "xsd:string". @sbarnum mentioned that this is invalid. The result in OWL currently looks like this:
```
core:DateTime a owl:Class ;
    rdfs:comment """A Datetime is a string representation of a specific date and time.
It has resolution of seconds and is always expressed in UTC timezone.
The specific format is one of the most commonly used ISO-8601 formats.
Format restriction: pattern: ^\\d\\d\\d\\d-\\d\\d-\\d\\dT\\d\\d:\\d\\d:\\d\\dZ$""" ;
    rdfs:subClassOf xsd:string ;
    ns0:term_status "Stable" .
```
As the whole point of these "classes" is to have them restricted via a specific pattern, I propose to change them to simple properties that must follow the pattern. That is, move and convert the SemVer, MediaType and DateTime md-files to the Properties folder. Summary and Description would be kept intact so that we lose no information in the process.

For the OWL this would mean to change for example the following snippet
```
[ sh:datatype core:DateTime ;
            sh:maxCount 1 ;
            sh:name "buildEndTime" ;
            sh:path build:buildEndTime ]
```
into 
```
[ sh:datatype xsd:string ;
            sh:maxCount 1 ;
            sh:name "buildEndTime" ;
            sh:path build:buildEndTime ;
            sh:pattern "^\\d\\d\\d\\d-\\d\\d-\\d\\dT\\d\\d:\\d\\d:\\d\\dZ$" ]
```

Please speak up if there are aspects to this that I haven't considered, else I'll open a PR for this.

**Comment by sbarnum** [comment]
Just a note that the above proposed SHACL change is appropriate for application to the NodeShape for the Class that the property applies to. The spec will also need the actual definitions of the properties themselves to also change to have 'rdf:type owl:DatatypeProperty ;' and 'rdfs:range xsd:string ;'

**Comment by davaya** [comment]
They aren't actually "strings that fit a pattern".  SemVer, or semantic version, has the semantics of three integers that must be followed - even if a SemVer is serialized as a string, it must sort as three integers, which strings don't do.  SPDX says follow the SemVer spec, which says:

> A normal version number MUST take the form X.Y.Z where X, Y, and Z are non-negative integers, and MUST NOT contain leading zeroes. X is the major version, Y is the minor version, and Z is the patch version. Each element MUST increase numerically. For instance: 1.9.0 -> 1.10.0 -> 1.11.0.

So the semantic, logical value of a SemVer isn't a string with a pattern, it must be a type with the specified semantic behavior regardless of how it is serialized.

**Comment by davaya** [comment]
Likewise, MediaType isn't a string with a pattern, it is a registry name followed by a registered type: https://www.iana.org/assignments/media-types/media-types.xhtml.

"blurf/farbo" isn't a valid media type; "image/jpg" is.

**Comment by davaya** [comment]
> Currently, [SemVer](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/SemVer.md), [MediaType](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/MediaType.md) and [DateTime](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/DateTime.md) are treated as classes that are subclasses of "xsd:string". @sbarnum mentioned that this is invalid.

As I mentioned in PR #377, it is not invalid to define SemVer, MediaType, and DateTime as Classes that are restricted versions of string.  Whether you choose to call them "subclasses" or not is hairsplitting, the point is that they can be used as types of properties just like all other types.

**Comment by davaya** [comment]
> Currently, [SemVer](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/SemVer.md), [MediaType](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/MediaType.md) and [DateTime](https://github.com/spdx/spdx-3-model/blob/main/model/Core/Classes/DateTime.md) are treated as classes that are subclasses of "xsd:string". @sbarnum mentioned that this is invalid.

As I mentioned in PR #377, it is not invalid to define SemVer, MediaType, and DateTime as Classes that are restricted versions of string.  Whether you choose to call them "subclasses" or not is hairsplitting, what matters is that they can be used as property types just like all other Classes in the model.

**Comment by goneall** [comment]
Fixed in #426
